Makefile

```diff
@@ -2,38 +2,57 @@ LISP ?= sbcl
 
 all: build
 
+# can use bespoke dir like 'QLDIR=~/nostandard/local-projects make'
+QLDIR ?= $(HOME)/quicklisp/local-projects
+
+# make will exit early if git clone errors b/c dir already exists
+define git-clone-pull =
+if test -d $(QLDIR)/$(notdir $1); then cd $(QLDIR)/$(notdir $1) && git pull; else git clone $1 $(QLDIR)/$(notdir $1); fi
+endef
+
+$(QLDIR)/asdf:
+	# 2024-08: building with older asdf fails
+	# unrecognized define-package keyword :LOCAL-NICKNAMES
+	# https://github.com/ciel-lang/CIEL/issues/58
+	mkdir -p $(QLDIR)
+	cd $(QLDIR) && \
+		curl -sL https://asdf.common-lisp.dev/archives/asdf-3.3.5.tar.gz | \
+		tar -xvzf - && \
+		mv asdf-3.3.5 asdf
+
 # Install some Quicklisp dependencies.
-ql-deps:
+ql-deps: $(QLDIR)/asdf
```

Comment:
I suggest that either we remove this automatic target dependency, either we check the asdf version. (I'll accept both, preference for the check).

Reply:
I removed the automatic download and added a version check.